front/src/modules/simulationResult/components/SpaceTimeChart/index.tsx

```diff
@@ -51,42 +67,75 @@ const CHART_MIN_HEIGHT = 250;
 
 export type SpaceTimeChartProps = {
   allowancesSettings?: AllowancesSettings;
-  dispatchUpdateDepartureArrivalTimes?: DispatchUpdateDepartureArrivalTimes;
-  dispatchUpdateChart?: DispatchUpdateChart;
-  dispatchUpdateMustRedraw?: DispatchUpdateMustRedraw;
-  dispatchUpdateSelectedTrainId?: DispatchUpdateSelectedTrainId;
-  updateTimePosition?: DispatchUpdateTimePositionValues;
   initialHeight?: number;
   inputSelectedTrain?: Train;
   selectedProjection?: OsrdSimulationState['selectedProjection'];
   simulation?: SimulationSnapshot;
   simulationIsPlaying?: boolean;
-  onOffsetTimeByDragging?: (trains: Train[], offset: number, timePosition: Date) => void;
   onSetBaseHeight?: (newHeight: number) => void;
   isDisplayed?: boolean;
 };
 
 export default function SpaceTimeChart(props: SpaceTimeChartProps) {
   const ref = useRef<HTMLDivElement>(null);
   const rndContainerRef = useRef<Rnd>(null);
+  const dispatch = useDispatch();
+
+  const allowancesSettingsFromStore = useSelector(getAllowancesSettings);
+  const inputSelectedTrainFromStore = useSelector(getSelectedTrain);
+  // implement selector for all selected trains ids
+  const selectedProjectionFromStore = useSelector(getSelectedProjection);
+  const simulationFromStore = useSelector(getPresentSimulation);
+  const simulationIsPlayingFromStore = useSelector(getIsPlaying);
+
+  const { updateTimePosition } = useChartSynchronizer();
+
+  const dispatchUpdateMustRedraw: DispatchUpdateMustRedraw = (newMustRedraw: boolean) => {
+    dispatch(updateMustRedraw(newMustRedraw));
+  };
+
+  const dispatchUpdateChart: DispatchUpdateChart = (chart: Chart | undefined) => {
```

Comment:
I'm not sure we want to be able to update the chart with undefined. Instead we could add a `if (chart)` in the useEffect.
```suggestion
  const dispatchUpdateChart: DispatchUpdateChart = (chart: Chart) => {
```